### PR TITLE
SO-6321: DSV simple type export should consider the characteristic type of properties

### PR DESCRIPTION
**Pull request by apeteri**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java [hunk 1]
```diff
@@ -34,44 +37,87 @@
...
+			BranchContext context, 
+			boolean includeInactiveMembers,
+			String refSetId
+		);
+		
+		ConceptStreamFactory DEFAULT = (expand, locales, context, includeInactiveMembers, refSetId) -> {
+			SnomedConceptSearchRequestBuilder builder = SnomedRequests.prepareSearchConcept()
+				.setLocales(locales)
+				.setExpand(expand)
+				.sortBy(Sort.fieldAsc(SnomedConceptDocument.Fields.ID))
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java]: Btw, this is the default sort field for all revision documents.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java [hunk 2]
```diff
@@ -34,44 +37,87 @@
...
+		
+		SnomedConcepts getAncestors(
+			List<ExtendedLocale> locales,
+			String ancestorId,
+			BranchContext context 
+		);
+		
+		AncestorCollector DEFAULT = (locales, ancestorId, context) -> {
+			return SnomedRequests.prepareSearchConcept()
+				.all()
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java:106]: Would it make sense to stream this as well?

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java (hunk 1 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java]: I see a `_default` sort here: https://github.com/b2ihealthcare/snow-owl/blob/5f2240d086432d66d32a8e441e062e8f7937ae43/commons/com.b2international.index/src/com/b2international/index/query/SortBy.java#L47-L55

Does the mapping mark `id` as the default field?

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java (hunk 2 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java:106]: It doesn't seem to be critical, this request is used to get up-to-date labels for type concept hierarchies (relationship attributes and description types), both of which are reasonably sized.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java (hunk 1 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java]: Yes, for Revision documents that's the default sort field: 
https://github.com/b2ihealthcare/snow-owl/blob/9.x/commons/com.b2international.index/src/com/b2international/index/revision/Revision.java#L106-L107

**Review by cmark — APPROVED** [review]
🥓

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java (hunk 2 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java:106]: Sorting removed in https://github.com/b2ihealthcare/snow-owl/pull/1349/commits/8701276e51aa71747ad40a4aa3700659151f40a3.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java (hunk 1 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/dsv/SnomedSimpleTypeRefSetDSVExporter.java]: Sorting removed in https://github.com/b2ihealthcare/snow-owl/pull/1349/commits/8701276e51aa71747ad40a4aa3700659151f40a3.

**Review by AAAlinaaa — APPROVED** [review]

**Pull request merged**